db/dao/attempt_dao.go

```diff
@@ -11,18 +11,21 @@ type attemptDao struct {
 	*DAO[entities.Attempt]
 }
 
-func (dao *attemptDao) UpdateDelivery(
-	ctx context.Context,
-	id string,
-	request *entities.AttemptRequest,
-	response *entities.AttemptResponse,
-	attemptAt time.Time,
-	status entities.AttemptStatus) error {
+type DeliveryResult struct {
+	Request   *entities.AttemptRequest
+	Response  *entities.AttemptResponse
+	AttemptAt time.Time
+	Status    entities.AttemptStatus
+	ErrorCode *entities.AttemptErrorCode
+}
+
+func (dao *attemptDao) UpdateDelivery(ctx context.Context, id string, result *DeliveryResult) error {
 	_, err := dao.update(ctx, id, map[string]interface{}{
-		"request":    request,
-		"response":   response,
-		"attempt_at": attemptAt.Unix(),
-		"status":     status,
+		"request":    result.Request,
+		"response":   result.Response,
+		"attempt_at": result.AttemptAt.Unix(),
```

Comment:
`Unix()` precision is not enough, it's only at the second level. It would be best if it could reach the millisecond level.

Reply:
Using Unix() just for keeping the unit consistency with the `created_at` and `updated_at`.